src/common/snippets/src/lowered/shape_inference/shape_infer_instances.cpp

```diff
@@ -0,0 +1,179 @@
+                    if (src[i] > dst[axis + i])
+                        return false;
+                }
+                success &= broadcast_merge_dim(dst[axis + i], dst[axis + i], src[i]);
+            }
+            return success;
+        }
+        default:
+            OPENVINO_THROW("Unsupported auto broadcast type: ", autob.m_type);
+    }
+    return false;
+}
+bool merge_into(IShapeInferSnippets::VectorDims& dst, const IShapeInferSnippets::VectorDims& src) {
+    auto merge_dim = [](size_t& dst, const size_t& d1, const size_t& d2) {
+        if (d1 == d2 || d1 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d2;
+        } else if (d2 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d1;
+        } else {
+            return false;
+        }
+        return true;
+    };
+    if (dst.size() != src.size())
+        return false;
+
+    bool success = true;
+    for (size_t i = 0; i < dst.size(); i++)
+        success &= merge_dim(dst[i], dst[i], src[i]);
+    return success;
+}
+} // namespace
+
+
+IShapeInferSnippets::Result
+entryNumpyBroadcasting::infer(const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes) {
+        size_t max_rank = 0;
+        size_t max_rank_idx = 0;
+        for (size_t i = 0; i < input_shapes.size(); ++i) {
+            auto item_rank = input_shapes[i].get().size();
+            if (item_rank > max_rank) {
+                max_rank = item_rank;
+                max_rank_idx = i;
+            }
+        }
+        auto output_shape = input_shapes[max_rank_idx].get();
+        // use NUMPY broadcast rule
+        for (size_t i = 0; i < input_shapes.size(); i++) {
+            if (i == max_rank_idx)
+                continue;
+
+            auto& input_shape = input_shapes[i].get();
+            const size_t offset = max_rank - input_shape.size();
+            for (size_t j = 0; j < input_shape.size(); ++j) {
+                if (input_shape[j] != output_shape[offset + j]) {
+                    if (output_shape[offset + j] == 1) {
+                        output_shape[offset + j] = input_shape[j];
+                    } else if (input_shape[j] != 1) {
+                        OPENVINO_THROW(input_shapes.size() == 1, "Got not broadcastable input shapes");
+                    }
+                }
+            }
+        }
+        return { { std::move(output_shape) }, ShapeInferStatus::success };
+}
+
+template<class BroadcastOP>
+BroadcastShapeInfer<BroadcastOP>::BroadcastShapeInfer(const std::shared_ptr<Node>& n) {
+        static_assert(std::is_base_of<snippets::op::BroadcastMove, BroadcastOP>() ||
+                      std::is_base_of<snippets::op::BroadcastLoad, BroadcastOP>(),
+                      "This ShapeInfer class could be used only for BroadcastMove and BroadcastLoad operations.");
+        const auto& broadcast = as_type_ptr<BroadcastOP>(n);
+        OPENVINO_ASSERT(ov::is_type<BroadcastOP>(n), "Invalid node passed to BroadcastShapeInfer.",
```

Comment:
Minor: can we just check `broadcast` on nullptr here?
```suggestion
        OPENVINO_ASSERT(broadcast, "Invalid node passed to BroadcastShapeInfer.",
```

Reply:
We sure can, thanks